src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -281,20 +304,7 @@ public ImmutableArray<U> ToDowncastedImmutable<U>()
         /// </summary>
         public ImmutableArray<T> ToImmutableAndFree()
```

Comment:
>ToImmutableAndFree [](start = 33, length = 18)

Is this function more expensive now, i.e. does more work than it used to?

Reply:
It appears yes. For builders with a capacity larger than 128 elements, there is a new call to `Clear` before it is discarded. i.e. the call to `Clear` was (unintentionally) hoisted outside this guard:

https://github.com/dotnet/roslyn/blob/a9609cc5a15f720242563a0271a45dc517830bb7/src/Dependencies/PooledObjects/ArrayBuilder.cs#L337

Reply:
➡️ Restored original implementation of `ToImmutableAndFree` with comment to prevent future regression.